docs/flow/modin/experimental/reshuffling_groupby.rst

```diff
@@ -0,0 +1,70 @@
+Reshuffling GroupBy
+"""""""""""""""""""
+
+The experimental GroupBy implementation utilizes Modin's reshuffling mechanism that gives an
+ability to build range partitioning over a Modin DataFrame.
+
+In order to enable/disable this new implementation you have to specify ``cfg.ExperimentalGroupbyImpl``
+:doc:`configuration variable: </flow/modin/config>`
+
+.. code-block:: ipython
+
+    In [4]: import modin.config as cfg; cfg.ExperimentalGroupbyImpl.put(True)
+
+    In [5]: # past this point Modin will always use the new reshuffling groupby implementation
+
+    In [6]: cfg.ExperimentalGroupbyImpl.put(False)
+
+    In [7]: # past this point Modin won't use reshuffling groupby implementation no more
+
+The reshuffling implementation appears to be quite efficient when compared to old TreeReduce and FullAxis implementations:
+
+.. code-block:: ipython
+
+    In [4]: import modin.pandas as pd; import numpy as np
```

Comment:
It would be nice to specify the number of cores or partitions at which such an acceleration can be obtained, in order to improve the reproducibility of the results by users.

Reply:
added a note regarding cpu